lib/kubernetes-deploy/runner_task.rb

```diff
@@ -0,0 +1,106 @@
+      @namespace = namespace
+      @kubeclient = build_v1_kubeclient(context)
+      @context = context
+    end
+
+    def run(task_template:, command:, args:, env_vars: [])
+      validate_configuration(task_template, args)
+
+      raw_template = get_template(task_template, namespace: @namespace)
+      rendered_template = build_pod_template(raw_template, command, args, env_vars)
+
+      KubernetesDeploy.logger.info("Starting task runner pod #{rendered_template.metadata.name}")
+      @kubeclient.create_pod(rendered_template)
+    end
+
+    private
+
+    def validate_configuration(task_template, args)
+      errors = []
+      if task_template.nil?
+        errors << "Task template name can't be nil"
+      end
+
+      if args.nil?
+        errors << "Args can't be nil"
+      end
+
+      raise FatalTaskRunError, "Configuration invalid: #{errors.join(', ')}" unless errors.empty?
+    end
+
+    def get_template(template_name, namespace:)
+      KubernetesDeploy.logger.info("Fetching task runner pod template: '#{template_name}' in namespace: '#{namespace}'")
+      config_map = @kubeclient.get_config_map(template_name, namespace: namespace)
+
+      if config_map.nil?
+        raise FatalTaskRunError, "No configmap found matching name=#{template_name} in namespace=#{namespace}"
+      end
+
+      raw = config_map.data.spec
+      template = YAML.load raw
+
+      Kubeclient::Pod.new(template)
+    rescue Psych::SyntaxError => e
+      KubernetesDeploy.logger.error(raw.to_hash)
+      raise FatalTaskRunError, "Task template #{template_name} cannot be parsed: #{e.message}"
+    end
+
+    def build_pod_template(base_template, command, args, env_vars)
+      KubernetesDeploy.logger.info("Rendering template for task runner pod")
+
+      rendered_template = base_template.dup
+      rendered_template.spec.containers
+        .first.args = args
+
+      container = rendered_template.spec.containers.find { |cont| cont.name == 'task-runner' }
+
+      raise FatalTaskRunError, "Pod spec does not contain a template container called 'task-runner'" if container.nil?
+
+      container.command = command
+
+      container.env += env_vars.map {|env| env.split('=')}.map do |key, value|
+        { name: key, value: value }
+      end
+
+      unique_name = rendered_template.metadata.name + "-" + SecureRandom.hex(8)
+
+      KubernetesDeploy.logger.warn("Name is too long, using '#{unique_name[0..62]}'") if unique_name.length > 63
+      rendered_template.metadata.name = unique_name[0..62]
+      rendered_template.metadata.namespace = @namespace
+
+      validate_pod_spec(rendered_template)
+
+      rendered_template
+    end
+
+    def validate_pod_spec(pod)
+      f = Tempfile.new(pod.metadata.name)
+      f.write pod.to_hash.to_json
+      f.close
+
+      _out, _err, status = run_kubectl("apply", "--dry-run", "-f", f.path, namespace: @namespace, context: @context)
+
+      unless status.success?
+        raise FatalTaskRunError, "Invalid pod spec"
```

Comment:
I think we should log the error message, otherwise it will be hard to debug and reproduce it.